test/conftest.py

```diff
@@ -24,4 +24,25 @@
 # THE SOFTWARE.
 import sys
 import os
+
+import pytest
+
+from apprise import NotifyBase
+from apprise.plugins.NotifyPushBullet import NotifyPushBullet
```

Comment:
I love the idea of the fixture, but I'm not sure if i agree with say `NotifyPushBullet` being hardcoded into it.  It means people who want to leverage this for new tests have to update this file as well as `test/test_plugin_newplugin.py`. 

 Up until now all 87 plugins have jsut one `test/text_plugin_abc.py` file.

that said...  Email, JSON, and XML plugins are a bit of an exception as i used them in many places to verify (config files, etc).  So you could technically call me out here, but i'd argue they're core functions that would never be decommissioned where something like PushBullet, or others may come and go over time (be bought out, merge, go bankrupt, etc) :)

Is there a way the `fixture` could take in the class name as an input so it's more reusable that way? The throttling is over-ridden in several other plugins. Once done, updating just the `NotifyBase` isn't enough. It needs an update to the parent class (As you did here).

Alternatively another great idea (who cares about the overhead as it's at test anyway) you just do a `forloop` over the plugins and set all of the parent request_rate to zero before the yield? Then no input is required? It also becomes a very reusable fixture?

Reply:
> The throttling is over-ridden in several other plugins. Once done, updating just the NotifyBase isn't enough. It needs an update to the parent class (As you did here).

Yeah, I recognized that. Thanks for the explanation.

> Alternatively another great idea (who cares about the overhead as it's at test anyway) you just do a `forloop` over the plugins and set all of the parent request_rate to zero before the yield? Then no input is required? It also becomes a very reusable fixture?

Hehe, a little bit brute-forcy, but why not?

> Is there a way the fixture could take in the class name as an input so it's more reusable that way?

The way to implement that would be to make a fixture return something which can be used to disable throttling. It can be anything, in this case a plain function or even a reference to another object instance will give you peace of mind for any flexbility you want to achieve here.

For example, given that we would provide a fixture called `throttler`, it would be used like `throttler.disable(NotifyPushBullet)`. I like that API, it is both explicit and easy to use. I think I will implement it this way, if you don't have any objections.

Reply:
No objections at all! Totally love that approach too.

Reply:
Oh, I just found that only #689 will bring in `pytest-mock`, that's why `no_throttling` here is still in its early form, while it has been rewritten to already use the `mocker` fixture later with d206c35c. Would you mind if I would improve this later, on #689 instead?

Reply:
I've improved the implementation with 9c2574fe, which is part of #689. I changed my mind about the `throttling` fixture, and provided a dedicated `no_throttling` fixture for `NotifyPushBullet` instead. No need to overcomplicate things.

> Do not abuse the "global" `no_throttling` fixture for specific notifiers. By using a module-scoped fixture with autouse, it will be even more convenient to use.

Reply:
Before the merge, is done; it appears the `NotifyPushBullet` class is still present in this code? Or am i looking at maybe the wrong branch?

Reply:
This spot has been improved on behalf of #689, which is a subsequent patch, right. The reason is because `no_throttling` has been rewritten to use `mocker` already.